src/ClientModel/SamplingDataPostProcessorTest.cpp

```diff
@@ -0,0 +1,721 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/container/flat_hash_set.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <vector>
+
+#include "ClientData/ModuleManager.h"
+#include "ClientModel/CaptureData.h"
+#include "ClientModel/SamplingDataPostProcessor.h"
+#include "OrbitBase/ThreadConstants.h"
+#include "capture_data.pb.h"
+
+using orbit_client_data::ModuleManager;
+using orbit_client_data::PostProcessedSamplingData;
+using orbit_client_data::SampledFunction;
+using orbit_client_data::ThreadSampleData;
+
+using orbit_client_protos::CallstackEvent;
+using orbit_client_protos::CallstackInfo;
+using orbit_client_protos::LinuxAddressInfo;
+
+using orbit_grpc_protos::CaptureStarted;
+
+using ::testing::ElementsAre;
+using ::testing::Eq;
+using ::testing::Pointwise;
+using ::testing::UnorderedElementsAre;
+
+namespace orbit_client_model {
+
+namespace {
+
+SampledFunction MakeSampledFunction(std::string name, std::string module_path, float exclusive,
+                                    float inclusive, uint64_t absolute_address) {
+  SampledFunction sampled_function;
+  sampled_function.name = std::move(name);
+  sampled_function.module_path = std::move(module_path);
+  sampled_function.exclusive = exclusive;
+  sampled_function.inclusive = inclusive;
+  sampled_function.absolute_address = absolute_address;
+  sampled_function.function = nullptr;
+  return sampled_function;
+}
+
+bool operator==(const SampledFunction& lhs, const SampledFunction& rhs) {
+  return lhs.name == rhs.name && lhs.module_path == rhs.module_path &&
+         lhs.exclusive == rhs.exclusive && lhs.inclusive == rhs.inclusive &&
+         lhs.absolute_address == rhs.absolute_address && lhs.function == rhs.function;
+}
+
+MATCHER_P(SampledFunctionEq, that, "") {
+  const SampledFunction& lhs = arg;
+  const SampledFunction& rhs = that;
+  return lhs == rhs;
+}
+
+// Need to redefine this as otherwise the standard std::vector comparison doesn't find
+// `bool operator==(const SampledFunction& lhs, const SampledFunction& rhs)` defined above.
+bool operator==(const std::vector<SampledFunction>& lhs, const std::vector<SampledFunction>& rhs) {
+  if (lhs.size() != rhs.size()) return false;
+  for (size_t i = 0; i < lhs.size(); ++i) {
+    if (!(lhs[i] == rhs[i])) return false;
+  }
```

Comment:
Nit: there is an std-algorithm for this use-case: https://en.cppreference.com/w/cpp/algorithm/lexicographical_compare

Reply:
I guess you meant `std::equal` as it's about equality and not order, but yes, done.